src/core/fs.cc

```diff
@@ -1714,53 +1714,85 @@ namespace SSC {
     });
   }
 
-  void Core::FS::mkdir (
+  static void mkdirRecursive(
+    const String& path,
+    int mode,
+    uv_loop_t* loop,
+    std::function<void(int)> callback
+  ) {
+    auto filename = path.c_str();
+
+    uv_fs_t stat_req;
+    uv_fs_stat(loop, &stat_req, filename, nullptr);
+
+    if (stat_req.result == 0) {
+        uv_fs_req_cleanup(&stat_req);
+        callback(0);
+        return;
+    }
+    uv_fs_req_cleanup(&stat_req);
+
+    size_t last_slash = path.rfind('/');
```

Comment:
we'll want to check for `\\` on Windows

Reply:
Whats the best pattern for doing that? Something like `path.sep` from node? A C++ macro?

Reply:
something like
```c++
#if defined(_WIN32)
size_t last_slash = path.find("\\");
#else
size_t last_slash = path.find("/");
#endif
```